addons/mail/static/src/messaging/entity/follower/follower.js

```diff
@@ -0,0 +1,175 @@
+odoo.define('mail.messaging.entity.Follower', function (require) {
+'use strict';
+
+const { registerNewEntity } = require('mail.messaging.entity.core');
+
+function FollowerFactory({ Entity }) {
+
+    class Follower extends Entity {
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * @return {string}
+         */
+        get name() {
+            if (this.channel) {
+                return this.channel.name;
+            }
+            return this.partner.name;
+        }
+
+        async remove() {
+            await this.env.rpc({
+                model: this.thread.model,
+                method: 'message_unsubscribe',
+                args: [
+                    [this.thread.id],
+                    this.partner ? [this.partner.id] : [],
+                    this.channel ? [this.channel.id] : [],
+                ],
+            });
+            this.delete();
+        }
+
+        async showSubtypes() {
+            const subtypesData = await this.env.rpc({
+                route: '/mail/read_subscription_data',
+                params: { follower_id: this.id },
+            });
+            this.unlink({ subtypes: null });
+            for (const data of subtypesData) {
+                const subtype = this.env.entities.FollowerSubtype.insert(data);
+                this.link({ subtypes: subtype });
+                if (data.followed) {
+                    this.link({ checkedSubtypes: subtype });
+                } else {
+                    this.unlink({ checkedSubtypes: subtype });
+                }
+            }
+            this.env.entities.Dialog.open('FollowerSubtypesEditor', { follower: this });
+        }
+
+        /**
+         * @param {mail.messaging.entity.FollowerSubtype} subtype
+         */
+        toggleSubtypeCheck(subtype) {
```

Comment:
I dislike toggle methods, can we just call unlink or link accordingly?

Reply:
I dont' know what @alexkuhn thinks about that but It keeps the logic in follower so the subtype just tell "ok I'm clicked, what's my new state ?". If you call unlink/link I think the state should be (re)computed in the subtype then. What do you guys think ?

Reply:
Usually prefer toggle just so that components stay as dumb as possible and intelligence is in entities, but when components are aware of current state and transitions like here, I kinda agree it shouldn't be a toggle.